packages/messaging/android/src/main/java/io/invertase/firebase/messaging/ReactNativeFirebaseMessagingModule.java

```diff
@@ -77,9 +78,10 @@ public void getInitialNotification(Promise promise) {
           if (messageId != null && initialNotificationMap.get(messageId) == null) {
             RemoteMessage remoteMessage = ReactNativeFirebaseMessagingReceiver.notifications.get(messageId);
             if (remoteMessage == null) {
-              remoteMessage = popRemoteMessageFromMessagingStore(messageId);
-            }
-            if (remoteMessage != null) {
+              promise.resolve(popRemoteMessageMapFromMessagingStore(messageId));
+              initialNotificationMap.put(messageId, true);
+              return;
+            } else {
```

Comment:
Hmm - I actually find this harder than it needs to be as a diff, and now there is duplicated logic.

It seems like the only change you *really* need to make is to change `popRemoteMessageFromMessagingStore` to be `popRemoteMessageMapFromMessagingStore` - can this entire diff block be reduced to that one change?

Reply:
It isn't *quite* that simple since we are still dealing with maps and remote messages, but your point is taken on the minimal changes.  Take a look, I think I cleaned it up nicely.